R/uvhydrograph-render.R

```diff
@@ -506,49 +512,44 @@ calculateLimitsAndSides <- function(primarySeriesList, uvInfo, refInfo, compInfo
 #' @param timeseries list of TS points relavant to primary plot
 #' @param name name of item being plotted
-#' @param primary_lbl label of primary time series
-#' @param reference_lbl label of refernce time series
-#' @param comp_lbl label of comparison time series
-#' @param dataSides named list of integers describing what side of the plot correspondingly named series goes on
-#' @param dataLimits named list of limits for name series
+#' @param label label of time series
+#' @param ylim y range that the time series covers
+#' @param dataSide optional for reference and comparison series, integer of what side the time series' y value is on
+#' @param comparisonOnIndependentAxes set to false if being plotted on the same axes as another
 #' @return named list of gsplot calls. The name is the plotting call to make, and it points to a list of config params for that call
-getPrimaryPlotConfig <- function(timeseries, name, primary_lbl, 
-    reference_lbl, comp_lbl, dataSides, dataLimits) {
+getPrimaryPlotConfig <- function(timeseries, name, label, 
+    ylim, dataSide=0, comparisonOnIndependentAxes=TRUE) {
   styles <- getUvStyles()
   
   x <- timeseries[['time']]
   y <- timeseries[['value']]
   
   compAxes <- TRUE
   compAnnotations <- TRUE
-  compLabel <- primary_lbl
   
-  if(dataSides[['comparison']] == 6){
+  if(comparisonOnIndependentAxes){
     compAxes <- FALSE
     compAnnotations <- FALSE
-    compLabel <- comp_lbl
-  } else if(dataSides[['comparison']] == 4 && (dataSides[['reference']] != 4)){
-    compLabel <- comp_lbl
-  }
+  } 
   
   plotConfig <- switch(name,
       corrected = list(
-          lines = append(list(x=x, y=y, ylim=dataLimits[['primary']], ylab=primary_lbl, legend.name=paste(styles[['corr_UV_lbl']], primary_lbl)), styles[['corr_UV_lines']])
+          lines = append(list(x=x, y=y, ylim=ylim, ylab=label, legend.name=paste(styles[['corr_UV_lbl']], label)), styles[['corr_UV_lines']])
           ),
       estimated = list(
-          lines = append(list(x=x, y=y, legend.name=paste(styles[['est_UV_lbl']], primary_lbl)), styles[['est_UV_lines']])
+          lines = append(list(x=x, y=y, legend.name=paste(styles[['est_UV_lbl']], label)), styles[['est_UV_lines']])
           ),
       uncorrected = list(
-          lines = append(list(x=x, y=y, legend.name=paste(styles[['uncorr_UV_lbl']], primary_lbl)), styles[['uncorr_UV_lines']])
+          lines = append(list(x=x, y=y, legend.name=paste(styles[['uncorr_UV_lbl']], label)), styles[['uncorr_UV_lines']])
           ),
       comparison = list(
-          lines = append(list(x=x, y=y, ylim=dataLimits[['comparison']], side=dataSides[['comparison']], axes=compAxes, ylab=compLabel, ann=compAnnotations, legend.name=comp_lbl), styles[['comp_UV_lines']])
+          lines = append(list(x=x, y=y, ylim=ylim, side=dataSide, axes=compAxes, ylab=label, ann=compAnnotations, legend.name=label), styles[['comp_UV_lines']])
```

Comment:
Did you mean to make all of these use the same label? Wouldn't this make the comparison series have the same legend name as the primary series?